src/java.management/share/classes/sun/management/ThreadImpl.java

```diff
@@ -408,21 +436,30 @@ protected void setThreadAllocatedMemoryEnabled(boolean enable) {
         }
     }
 
+    /**
+     * Returns an array of thread identifiers for the threads in the given
+     * array. Returns {@code null} if {@code threads} is null or the array
+     * of threads only includes carrier threads.
+     */
+    private long[] threadsToIds(Thread[] threads) {
+        if (threads != null) {
+            long[] tids = Stream.of(threads)
+                    .filter(t -> !(t instanceof jdk.internal.misc.CarrierThread))
```

Comment:
Returning an array of thread IDs of just the platform threads is good.   The javadoc needs to be updated to say "Returns an array of thread identifiers for the platform threads...."

Reply:
I think you mean the comment on the private method. Yes, that could be clearer that it just returns platform threads.